### PR TITLE
Cost Estimate of a Hydrogen Decrepitation Furnace

### PR DESCRIPTION
**Pull request by Akintomiwaojo**
## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.md file
   at the top level of this directory.

2. I represent I am authorized to make the contributions and grant the license. If my employer has
   rights to intellectual property that includes these contributions, I represent that I have
   received permission to make contributions and grant the required license on behalf of that
   employer.

### CODE REVIEW

File: src/prommis/HD_CE.py [hunk 1]
```diff
@@ -0,0 +1,344 @@
+from pyomo.environ import ConcreteModel, Constraint, Param, Var, units,PositiveReals, log
+from pyomo.environ import units as pyunits
+from idaes.core import FlowsheetBlock,UnitModelCostingBlock,UnitModelBlock
+from idaes.models.costing.SSLW import SSLWCosting, SSLWCostingData, VesselMaterial
+from idaes.core.solvers import get_solver
+from math import pi
```
Comment (andrewlee94) [inline src/prommis/HD_CE.py:6]: For consistency, you should get do:

```
from idaes.core.util.constants import Constants
```

You can then get `pi` as `Cosntants.pi` (along with a number of other useful constants).

File: src/prommis/HD_CE.py [hunk 2]
```diff
@@ -0,0 +1,344 @@
+from pyomo.environ import ConcreteModel, Constraint, Param, Var, units,PositiveReals, log
+from pyomo.environ import units as pyunits
```
Comment (andrewlee94) [inline src/prommis/HD_CE.py:2]: Combine this with the line above, as they all come from the same place.

File: src/prommis/HD_CE.py [hunk 3]
```diff
@@ -0,0 +1,344 @@
...
+from idaes.core.solvers import get_solver
+from math import pi 
+
+
+class HDFurnaceCostEstimator:
+
+    """Cost Estimation of an Hydrogen Decrepitation Furnace"""
+
+    def __init__(self,
+                ramp_up_time = 20,                                                  #Ramp Up time (in minutes)
```
Comment (andrewlee94) [inline src/prommis/HD_CE.py:14]: Standardize comment formatting - some are missing a space after the `#` . also, to fit our standard line length limitations, I suggest only using a double space before the in-line comment.

I will also note you are presuming units here which will limit generality later.

File: src/prommis/HD_CE.py [hunk 4]
```diff
@@ -0,0 +1,344 @@
+from pyomo.environ import ConcreteModel, Constraint, Param, Var, units,PositiveReals, log
```
Comment (andrewlee94) [inline src/prommis/HD_CE.py:1]: This is not the right place for this file. If it is costing related then it should go in a costing folder somewhere.

File: src/prommis/HD_CE.py [hunk 5]
```diff
@@ -0,0 +1,344 @@
...
+
+        vessel_weight = self.C_Ins1()[0]+self.C_Metal1()[0]+self.C_Ins2()[0]+self.C_Metal2()[0]
+        vessel_volume = self.C_Metal2()[4]*35.315
+        vessel_density = vessel_weight/vessel_volume
+        transportation_cost = (10+(0.57*vessel_density))*self.labor_rate 
+        inspection_cleaning_and_testing = (50+(1.892*self.C_Metal2()[3]))*self.labor_rate
+
+        CAPEX = self.C_Ins1()[1] + self.C_Metal1()[1] + self.C_Ins2()[1]+ self.C_Metal2()[1] + self.transformer_cost + self.temperature_controller_price+self.heating_coil() + transportation_cost + inspection_cleaning_and_testing                                             
+        return CAPEX
+
```
Comment (andrewlee94) [inline src/prommis/HD_CE.py:340]: Excess line breaks at end of file.

File: src/prommis/HD_CE.py [hunk 6]
```diff
@@ -0,0 +1,344 @@
...
+        temperature_pressure_controller_price - N480D PID Temperature controller
+        """
+
+        vessel_weight = self.C_Ins1()[0]+self.C_Metal1()[0]+self.C_Ins2()[0]+self.C_Metal2()[0]
+        vessel_volume = self.C_Metal2()[4]*35.315
+        vessel_density = vessel_weight/vessel_volume
+        transportation_cost = (10+(0.57*vessel_density))*self.labor_rate 
+        inspection_cleaning_and_testing = (50+(1.892*self.C_Metal2()[3]))*self.labor_rate
+
+        CAPEX = self.C_Ins1()[1] + self.C_Metal1()[1] + self.C_Ins2()[1]+ self.C_Metal2()[1] + self.transformer_cost + self.temperature_controller_price+self.heating_coil() + transportation_cost + inspection_cleaning_and_testing
```
Comment (andrewlee94) [inline src/prommis/HD_CE.py:338]: I am somewhat uncertain as to what the purpose of this file is - for the most part this is just doing Python operations which does not fit in with the rest of the code which works builds Pyomo models. Could you clarify the purpose of this code a bit more?

File: src/prommis/leaching/tests/test_leach_flowsheet.py [hunk 7]
```diff
@@ -1,19 +1,19 @@
-import pytest
-
 from pyomo.environ import (
-    assert_optimal_termination,
     SolverFactory,
     TransformationFactory,
+    assert_optimal_termination,
```
Comment (andrewlee94) [inline src/prommis/leaching/tests/test_leach_flowsheet.py:4]: Why is this file being edited here? These changes are just ordering of imports, and the original import order is more correct.

File: src/prommis/uky/costing/ree_plant_capcost.py [hunk 8]
```diff
@@ -35,17 +35,20 @@
 import textwrap
 from sys import stdout
 
+from pyomo.common.dependencies import attempt_import
```
Comment (andrewlee94) [inline src/prommis/uky/costing/ree_plant_capcost.py:38]: Similarly, these changes seem to be unrelated to this PR.

File: src/prommis/uky/costing/tests/test_ree_costing.py [hunk 9]
```diff
@@ -16,23 +16,49 @@
 """
 
 import pyomo.environ as pyo
-from pyomo.environ import check_optimal_termination
+from pyomo.common.dependencies import attempt_import
```
Comment (andrewlee94) [inline src/prommis/uky/costing/tests/test_ree_costing.py:19]: Again, at first glance this does not appear to be related to this PR.

**Review by andrewlee94 — CHANGES_REQUESTED** [review]
A few questions, mostly about what this PR is meant to be adding. There are a number of files being changed that don;t look like they are related to hydrogen decrepitation, and the one file that is about the hydrogen decripitation appears to be mostly Python code and in the wrong place.

**Comment by bpaul4** [comment]
@Akintomiwaojo thank you for opening this. Before @andrewlee94's addressing comments above, the branch is very out-of-date and should be updated. Please resolve any merge conflicts when doing so.

**Comment by Akintomiwaojo** [comment]
As suggested, I will push this to the PrOMMiS workspace.

**Pull request closed without merging**